django/db/models/indexes.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.core.exceptions import ValidationError
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# The max length of the names of the indexes (restricted to 30 due to Oracle)
+MAX_NAME_LENGTH = 30
+
+
+class Index(object):
+
+    index_type = 'idx'
+
+    def __init__(self, *fields, **kwargs):
```

Comment:
I don't think we should hijack `*args` for fields, a `fields` kwarg makes more sense to me.

`model` should be assigned by `Options` just like `Field.model` is.

What do you think of the `name`, `db_name` approach suggested above? In this case the signature should look like `(name, fields, db_name=None, **options)` and there would be no need to validate `name`.